hal/targets/hal/TARGET_ONSEMI/TARGET_NCS36510/spi_api.c

```diff
@@ -0,0 +1,204 @@
+/**
+ *******************************************************************************
+ * @file    spi_api.c
+ * @brief Implementation of a sleep functionality
+ * @internal
+ * @author  ON Semiconductor
+ * $Rev:  0.1 $
+ * $Date: 02-05-2016 $
+ ******************************************************************************
+ * @copyright (c) 2015 ON Semiconductor. All rights reserved.
+ * ON Semiconductor is supplying this software for use with ON Semiconductor
+ * processor based microcontrollers only.
+ *
+ * THIS SOFTWARE IS PROVIDED "AS IS".  NO WARRANTIES, WHETHER EXPRESS, IMPLIED
+ * OR STATUTORY, INCLUDING, BUT NOT LIMITED TO, IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE APPLY TO THIS SOFTWARE.
+ * ON SEMICONDUCTOR SHALL NOT, IN ANY CIRCUMSTANCES, BE LIABLE FOR SPECIAL,
+ * INCIDENTAL, OR CONSEQUENTIAL DAMAGES, FOR ANY REASON WHATSOEVER.
+ * @endinternal
+ *
+ * @ingroup spi_api
+ *
+ * @details
+ * SPI implementation
+ *
+ */
+#if DEVICE_SPI
+#include "spi.h"
+#include "PeripheralPins.h"
+#include "objects.h"
+#include "spi_api.h"
+#include "mbed_assert.h"
+#include "memory_map.h"
+#include "spi_ipc7207_map.h"
+#include "crossbar.h"
+#include "clock.h"
+#include "cmsis_nvic.h"
+
+
+#define SPI_FREQ_MAX			4000000
+#define SPI_ENDIAN_LSB_FIRST	0
+#define SPI_MASTER_MODE			1
+#define SPI_SLAVE_MODE			0
+
+void spi_init(spi_t *obj, PinName mosi, PinName miso, PinName sclk, PinName ssel)
+{
+	fSpiInit(obj, mosi, miso, sclk, ssel);
+}
+void spi_free(spi_t *obj)
+{
+	fSpiClose(obj);
+}
+
+void spi_format(spi_t *obj, int bits, int mode, int slave)
+{
+	if(slave)
+	{/* Slave mode */
+		obj->membase->CONTROL.BITS.MODE = SPI_SLAVE_MODE;
+	}
+	else
+	{/* Master mode */
+		obj->membase->CONTROL.BITS.MODE = SPI_MASTER_MODE;
+	}
+	obj->membase->CONTROL.BITS.WORD_WIDTH = bits >> 0x4; 			/* word width */
+	obj->membase->CONTROL.BITS.CPOL 	  = mode >> 0x1; 			/* CPOL  */
+	obj->membase->CONTROL.BITS.CPHA 	  = mode & 0x1; 			/* CPHA */
+
+	obj->membase->CONTROL.BITS.ENDIAN 	  = SPI_ENDIAN_LSB_FIRST;	/* Endian TODO ARM needs to confirm? */
+}
+
+void spi_frequency(spi_t *obj, int hz)
+{
+	/* TODO if the frequency is outside the allowable range, set it to the max */
+	if(hz > SPI_FREQ_MAX)
+	{
+		hz = SPI_FREQ_MAX;
+	}
+	obj->membase->FDIV = ((fClockGetPeriphClockfrequency() / hz) >> 1) - 1;
+}
+
+int  spi_master_write(spi_t *obj, int value)
+{
+	return(fSpiWriteB(obj, value));
+}
+
+int  spi_busy(spi_t *obj)
+{
+	return(obj->membase->STATUS.BITS.XFER_IP);
+}
+
+uint8_t spi_get_module(spi_t *obj)
+{
+	if(obj->membase == SPI1REG)
+	{
+		return 0; /* UART #1 */
+	}
+	else if(obj->membase == SPI2REG)
+	{
+		return 1; /* UART #2 */
+	}
+	else
+	{
+		return 2; /* Invalid address */
+	}
+}
+
+#if DEVICE_SPI_ASYNCH /* TODO Not implemented yet */
+
+void spi_master_transfer(spi_t *obj, void *tx, size_t tx_length, void *rx, size_t rx_length, uint32_t handler, uint32_t event, DMAUsage hint)
+{
+
+	uint32_t i; int ndata = 0;
```

Comment:
Can you please fix the alignment in files (tabs used?) and the style - make it consistent? Follow https://developer.mbed.org/teams/SDK-Development/wiki/mbed-sdk-coding-style for HAL implementation please.